internal/pkg/workspace/workspace.go

```diff
@@ -202,21 +250,25 @@ func (ws *Workspace) ReadWorkloadManifest(mftDirName string) (WorkloadManifest,
 	return mft, nil
 }
 
-// ReadPipelineManifest returns the contents of the pipeline manifest under copilot/pipeline.yml.
-func (ws *Workspace) ReadPipelineManifest() ([]byte, error) {
-	pmPath, err := ws.pipelineManifestPath()
-	if err != nil {
-		return nil, err
-	}
-	manifestExists, err := ws.fsUtils.Exists(pmPath)
-
+// ReadPipelineManifest returns the contents of the pipeline manifest under the given path.
+func (ws *Workspace) ReadPipelineManifest(path string) (*manifest.PipelineManifest, error) {
+	manifestExists, err := ws.fsUtils.Exists(path)
 	if err != nil {
 		return nil, err
 	}
 	if !manifestExists {
 		return nil, ErrNoPipelineInWorkspace
 	}
-	return ws.read(pipelineFileName)
+	data, err := ws.fsUtils.ReadFile(path)
+	if err != nil {
+		log.Infof("Unable to read pipeline manifest file at '%s'", path)
+		return nil, nil
+	}
```

Comment:
hmm I think it makes sense for `ReadPipelineManifest` to return an error if the `path` is invalid and it should be up to the client code to skip the error or not.